src/components/clusterWizard/ClusterDetails.tsx

```diff
@@ -0,0 +1,273 @@
+import { ClusterDetailsValues } from '../../types/clusters';
+import { getClusters, patchCluster } from '../../api/clusters';
+import { getErrorMessage, handleApiError } from '../../api/utils';
+import { updateCluster } from '../../features/clusters/currentClusterSlice';
+import { useDispatch } from 'react-redux';
+import { AlertsContext } from '../AlertsContextProvider';
+import ClusterWizardContext from './ClusterWizardContext';
+import Alerts from '../ui/Alerts';
+import CheckboxField from '../ui/formik/CheckboxField';
+import { getManagedDomains } from '../../api/domains';
+import ToolbarText from '../ui/Toolbar/ToolbarText';
+
+type ClusterDetailsFormProps = {
+  cluster: Cluster;
+  pullSecret: string;
+  managedDomains: ManagedDomain[];
+};
+
+const ClusterDetailsForm: React.FC<ClusterDetailsFormProps> = ({
+  cluster,
+  pullSecret,
+  managedDomains,
+}) => {
+  const { alerts, addAlert, clearAlerts } = React.useContext(AlertsContext);
+  const { setCurrentStepId } = React.useContext(ClusterWizardContext);
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const nameInputRef = React.useRef<HTMLInputElement>();
+  React.useEffect(() => {
+    nameInputRef.current?.focus();
+  }, []);
+  const { name, openshiftVersion = '', pullSecretSet, baseDnsDomain } = cluster;
+
+  const initialValues = {
+    name,
+    openshiftVersion,
+    pullSecret,
+    baseDnsDomain,
+    useRedHatDnsService:
+      !!baseDnsDomain && managedDomains.map((d) => d.domain).includes(baseDnsDomain),
+  };
+
+  const validationSchema = React.useCallback(
+    () =>
+      Yup.object({
+        name: nameValidationSchema,
+        openshiftVersion: Yup.string().required('Required'),
+        pullSecret: pullSecretSet
+          ? validJSONSchema
+          : validJSONSchema.required('Pull secret must be provided.'),
+        baseDnsDomain: dnsNameValidationSchema(initialValues.baseDnsDomain),
+      }),
+    [],
+  );
+
+  const handleSubmit = async (
+    values: ClusterDetailsValues,
+    formikActions: FormikHelpers<ClusterDetailsValues>,
+  ) => {
+    clearAlerts();
+
+    // async validation for cluster name - run only on submit
+    // TODO(jtomasek): Update this to validate combination of cluster name + dns domain
+    try {
+      const { data: clusters } = await getClusters();
+      const names = clusters.map((c) => c.name).filter((n) => n !== cluster.name);
+      if (names.includes(values.name)) {
+        return formikActions.setFieldError('name', `Name "${values.name}" is already taken.`);
+      }
+    } catch (e) {
+      captureException(e, 'Failed to perform unique cluster name validation.');
+    }
+
+    try {
+      const params = _.omit(values, ['useRedHatDnsService']);
+
+      if (pullSecretSet) {
+        delete params.pullSecret;
+      }
+
+      const { data } = await patchCluster(cluster.id, params);
+      dispatch(updateCluster(data));
+      setCurrentStepId('cluster-configuration');
```

Comment:
I think we should check relevant backend validations again before transitioning

Reply:
Agreed, however I am not sure if there any that we should check for this step?

Reply:
Just for consistency. If nothing else to have it among the transition functions, so we can think about all of them together.
Can be follow-up.